attributes/zzz_system_tuning.rb

```diff
@@ -0,0 +1,167 @@
+#
+# Cookbook Name:: hadoop
+# Attribute:: zzz_system_tuning
+#
+# Copyright © 2013-2015 Cask Data, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This attributes file sets up the port reservations for sysctl for COOK-79. Since these values must be compiled into
+# a single value, everything is done in this one file, and is done at Chef compile time, versus in a recipe. This is
+# to attempt to ensure that the cookbook only gets a single value for this setting on any one cluster.
+#
+# The source for these ports is http://private-repo-1.hortonworks.com.s3.amazonaws.com/HDPDocuments/HDP2/HDP-2.0.9.0/bk_using_Ambari_book/content/reference_chap2_2x.html
```

Comment:
Yeah, that ended up not being my source. I went to the *-default.xml files or docs for some of it as that source wasn't comprehensive and sometimes incorrect.